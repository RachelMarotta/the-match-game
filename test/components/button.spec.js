// importamos la funcion que vamos a testear
import Button from '../../src/components/button.js';

describe('Button', () => {
  it('should have the correct structure', () => {
    const element = document.createElement('div');
    element.innerHTML = Button({ title: 'Text', id: 1 });

    expect(element).toMatchSnapshot();
  });

  it('should call handleClick on button is clicked', () => {
    const temp = global.button.handleClick;
    global.button.handleClick = jest.fn();

    const element = document.createElement('div');
    element.innerHTML = Button({ title: 'Text', id: 1 });

    const button = element.querySelector('button');
    button.click();

    expect(global.button.handleClick.mock.calls).toHaveLength(1);
    global.button.handleClick = temp;
  });

  describe('handleClick', () => {
    it('should return a text with the id', () => {
      expect(global.button.handleClick('🍌')).toBe('Esse é o meu botão 🍌');
    });
  });
});
