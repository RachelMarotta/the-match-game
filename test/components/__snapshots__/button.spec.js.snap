// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have the correct structure 1`] = `
<div>
  
    
  <button
    class="primary-button"
    onclick="button.handleClick('1')"
  >
    Text
  </button>
  
  
</div>
`;
